### ISSUE TITLE
Class ClassifierAutoencoder missing

### ISSUE THREAD

**Issue by K-Coconut** [davidireland3/LeNSE#5]
Hi,

Class "ClassifierAutoencoder" can not be found at network.py
https://github.com/davidireland3/LeNSE/blob/31d117ced1cf7b4cf915c7cff29c343b37dedb2d/IM/autoencoder.py#L7

Wnat's the implementation for ClassifierAutoencoder in the paper?
I try to implement a ClassifierAutoencoder as below, is it the way similar to how you implemented it?

```python
class ClassifierAutoencoder(nn.Module):
    def __init__(self, input_size, num_layers):
        super(ClassifierAutoencoder, self).__init__()
        self.num_layers = num_layers
        self.input = [nn.Linear(input_size, input_size) for _ in range(num_layers - 1)] + [nn.Linear(input_size, 2)]
        self.input = nn.Sequential(*self.input)
        self.output = nn.Linear(2, input_size)

    def forward(self, x, embedding=False):
        x = self.input(x)
        if embedding:
            return x
        x = self.output(x)
        return x
```

**Comment by davidireland-iso** [comment]
Yes, the autoencoder is just for visualisation. If I remember correctly, the classifier autoencoder was because the inputs were a different size to the InfoNCE loss. To be honest, these were never meant to make it to the final repo. I’m on holiday at the moment but if you still need help by the time I am back I can try and find the code I used, otherwise just any (non-variations) autoencoder should do the trick, or indeed any dimensionality reduction technique.
